### ISSUE TITLE
[10.0] sale.order method action_cancel multi is broken

### ISSUE THREAD

**Issue by oerp-odoo** [odoo/odoo#18876]
Impacted versions:
10.0
Steps to reproduce:
Call `action_cancel` method on multiple sale.order records.
Current behavior (extended in `sale_stock` module):

```python
    @api.multi
    def action_cancel(self):
        self.order_line.mapped('procurement_ids').cancel()
        return super(SaleOrder, self).action_cancel()
```
As you can see this extension, breaks `multi` call, because if `self` has more than one record, it will crash.

Expected behavior:
It should be something like this.

```python
    @api.multi
    def action_cancel(self):
        self.mapped('order_line').mapped('procurement_ids').cancel()
        return super(SaleOrder, self).action_cancel()
```

**Comment by Yenthe666** [comment]
@mart-e do you agree with this? I'm happy to make a PR for this if you wish.

**Comment by mart-e** [comment]
@oerp-odoo thanks for the report and proposed fix. Do not hesitate to submit it directly in a PR if you already have the patch.

@Yenthe666 thanks but the time consuming part was to review the patch, not to write it (already given)

Fixed in 9.0 at 85a360eeaec84524a4a5180ec06af49fb564d465

**Comment by Yenthe666** [comment]
Great job!
